### PR TITLE
Adding taskcluster authentication

### PR DESCRIPTION
**Pull request by martiansideofthemoon**
ref #413 

This change is [Reviewable](https://reviewable.io/reviews/mozilla/mozilla_ci_tools/462)

### CODE REVIEW

File: test/taskcluster/test_taskcluster.py [hunk 1]
```diff
@@ -63,6 +64,9 @@ def test_check_valid_graph(self, good_graph1):
         # Similar to the previous test, but with the correct owner field
         validate_graph(good_graph1)
 
+    def test_authenticate(self):
+        authenticate()
```
Comment (armenzg) [inline test/taskcluster/test_taskcluster.py]: We're going to need some mocking of the response from the server.

Here's a blog post I've written on what to do to figure out what mocks or @patches to add:
http://armenzg.blogspot.ca/2016/04/improving-how-i-write-python-tests.html

**Comment by armenzg** [comment]
Did you get to see the browser open up and ask you to authenticate?

File: mozci/taskcluster/tc.py [hunk 2]
```diff
@@ -333,3 +333,7 @@ def validate_graph(graph):
 
     # validate() does not return a value if valid, thus, not keeping track of it.
     validate(instance=graph, schema=schema, format_checker=FormatChecker())
+
+
+def authenticate():
```
Comment (armenzg) [inline mozci/taskcluster/tc.py:338]: Could you please add some comments as to what this does?

**Comment by armenzg** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/mozilla_ci_tools/462)_

**Comment by martiansideofthemoon** [comment]
@armenzg **r**?

**Pull request merged**